src/backend/distributed/utils/attribute.c

```diff
@@ -212,20 +210,43 @@ AttributeQueryIfAnnotated(const char *query_string, CmdType commandType)
 			Datum jsonbDatum = DirectFunctionCall1(jsonb_in, PointerGetDatum(annotation));
 
 			text *tenantIdTextP = ExtractFieldTextP(jsonbDatum, "tId");
+			char *tenantId = NULL;
 			if (tenantIdTextP != NULL)
 			{
-				char *tenantId = UnescapeCommentChars(text_to_cstring(tenantIdTextP));
-				strncpy_s(attributeToTenant, MAX_TENANT_ATTRIBUTE_LENGTH, tenantId,
-						  MAX_TENANT_ATTRIBUTE_LENGTH - 1);
+				tenantId = UnescapeCommentChars(text_to_cstring(tenantIdTextP));
 			}
 
-			colocationGroupId = ExtractFieldInt32(jsonbDatum, "cId",
-												  INVALID_COLOCATION_ID);
+			int colocationId = ExtractFieldInt32(jsonbDatum, "cId",
+												 INVALID_COLOCATION_ID);
+
+			AttributeTask(tenantId, colocationId, commandType);
 		}
 	}
-	else
+}
+
+
+/*
+ * AttributeTask assigns the given attributes of a tenant and starts a timer
+ */
+void
+AttributeTask(char *tenantId, int colocationId, CmdType commandType)
+{
+	if (tenantId == NULL || colocationId == INVALID_COLOCATION_ID)
+	{
+		return;
+	}
+
+	colocationGroupId = colocationId;
+	strncpy_s(attributeToTenant, MAX_TENANT_ATTRIBUTE_LENGTH, tenantId,
+			  MAX_TENANT_ATTRIBUTE_LENGTH - 1);
+	attributeCommandType = commandType;
+
+	if (MultiTenantMonitoringLogLevel != CITUS_LOG_LEVEL_OFF)
 	{
-		strcpy_s(attributeToTenant, sizeof(attributeToTenant), "");
+		ereport(NOTICE, (errmsg(
+							 "attributing query to tenant: %s, colocationGroupId: %d",
+							 quote_literal_cstr(attributeToTenant),
+							 colocationGroupId)));
```

Comment:
we removed these types of logs everywhere else in https://github.com/citusdata/citus/pull/6762, since they only seemed to be useful for debugging. Let's remove it here too.